src/mol-repr/volume/isosurface.ts

```diff
@@ -158,12 +164,23 @@ namespace VolumeIsosurfaceTexture {
             throw new Error('volume too large for gpu isosurface extraction');
         }
 
+        const dataType = props.gpuDataType === 'halfFloat' && !webgl.extensions.textureHalfFloat ? 'float' : props.gpuDataType;
+
+        if (volume._propertyData[name]?.dataType !== dataType) {
+            volume.customProperties.remove(descriptor);
+            delete volume._propertyData[name];
+        }
+
         if (!volume._propertyData[name]) {
```

Comment:
```suggestion
        if (volume._propertyData[name]?.dataType !== dataType) {
            // this might be redundant, up to you if you wanna keep it
            volume.customProperties.assets(descriptor);
        }

        if (!volume._propertyData[name] || volume._propertyData[name]?.dataType !== dataType) {
```

And remove the newly added `remove` function?

Reply:
right, simplified it